pkg/vdri/key/resolver.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package key
+
+import (
+	"bytes"
+	"fmt"
+	"regexp"
+
+	"github.com/btcsuite/btcutil/base58"
+
+	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
+	vdriapi "github.com/hyperledger/aries-framework-go/pkg/framework/aries/api/vdri"
+)
+
+// Read expands did:key value to a DID document.
+func (v *VDRI) Read(didKey string, opts ...vdriapi.ResolveOpts) (*did.Doc, error) {
+	parsed, err := did.Parse(didKey)
+	if err != nil {
+		return nil, err
+	}
+
+	if !isValidMethodID(parsed.MethodSpecificID) {
+		return nil, fmt.Errorf("invalid did:key method ID: %s", parsed.MethodSpecificID)
+	}
+
+	pubKey, err := pubKeyFromFingerprint(parsed.MethodSpecificID)
+	if err != nil {
+		// TODO: cover path with test
+		return nil, err
+	}
+
+	return createDoc(pubKey)
+}
+
+func isValidMethodID(id string) bool {
+	r := regexp.MustCompile(`(z)([a-km-zA-HJ-NP-Z1-9]+)`)
+	return r.MatchString(id)
+}
+
+func pubKeyFromFingerprint(fingerprint string) ([]byte, error) {
+	// did:key:MULTIBASE(base58-btc, MULTICODEC(public-key-type, raw-public-key-bytes))
+	// https://w3c-ccg.github.io/did-method-key/#format
+	mc := base58.Decode(fingerprint[1:]) // skip leading "z"
```

Comment:
Are you validating the prefix character somewhere?

Reply:
yep, it's done in `isValidMethodDID` func before call to `pubKeyFromFingerprint`